lib/rubocop/ast/node/block_node.rb

```diff
@@ -153,15 +155,15 @@ def void_context?
 
       private
 
-      # Numbered arguments of this `numblock`.
       def numbered_arguments
-        return [].freeze unless numblock_type?
-
         max_param = children[1]
         1.upto(max_param).map do |i|
           ArgNode.new(:arg, [:"_#{i}"])
         end.freeze
       end
+
+      IMPLICIT_IT_ARGUMENT = [ArgNode.new(:arg, [:it])].freeze
+      private_constant :IMPLICIT_IT_ARGUMENT
```

Comment:
Can you move this constant definition above the method definition?

```diff
       include MethodIdentifierPredicates

+      IMPLICIT_IT_ARGUMENT = [ArgNode.new(:arg, [:it])].freeze
+      private_constant :IMPLICIT_IT_ARGUMENT
       VOID_CONTEXT_METHODS = %i[each tap].freeze
       private_constant :VOID_CONTEXT_METHODS
```

Reply:
And `IT_BLOCK_ARGUMENT` might be a clearer name.

Reply:
Thank you for explaining the intention. However, constants are generally expected to be at the top, so please move it accordingly. This will also maintain consistency with how existing constants are handled in this file.